torch/autograd/gradcheck.py

```diff
@@ -419,6 +430,25 @@ def vjp_fn(grad_output):
     return jacobians
 
 
+def get_fast_analytical_jacobians(inputs, outputs, nondet_tol, check_grad_dtypes, all_v, all_u_dense):
+    reduced_jacobians: List[List[torch.Tensor]] = []
+    for output, v in zip(outputs, all_v):
+        jacobian_cols = check_analytical_jacobian_attributes(inputs, output, nondet_tol, check_grad_dtypes,
+                                                             fast_mode=True, v=v)
+        jacobian_scalars: List[torch.Tensor] = []
+        for vJ, u in zip(jacobian_cols, all_u_dense):
+            vJ = vJ.T.squeeze(0)
```

Comment:
This squeeze looks very arbitrary. Why do you need it?

Reply:
vJ is a 2-d tensor because this allows us to reuse the error checking logic in the slow case as-is

Reply:
Ok, Maybe an extra comment would help.